### ISSUE TITLE
Replay of http2 flows seems to be broken

### ISSUE THREAD

**Issue by rozenbom** [mitmproxy/mitmproxy#4318]
#### Problem Description
When http2 flow is replayed "host"-header is missed which leads to a server error. The root cause is probably the following:
1. During flow construction ":authority"-header is removed: https://github.com/mitmproxy/mitmproxy/blob/880cea66bba2bd126b353d2822466300861f1289/mitmproxy/proxy/protocol/http2.py#L511
2. During replay a value for "host"-header is sourced from ":authority"-header: https://github.com/mitmproxy/mitmproxy/blob/880cea66bba2bd126b353d2822466300861f1289/mitmproxy/addons/clientplayback.py#L208.

#### Steps to reproduce the behavior:
1. Perform a http2 request
`curl --http2 https://httpbin.org/get -k -x localhost:9099`
2. Replay the flow.
3. Observe 400-error.

#### System Information
```
Mitmproxy: 5.3.0
Python:    3.8.6
OpenSSL:   OpenSSL 1.1.1h  22 Sep 2020
Platform:  macOS-10.15.7-x86_64-i386-64bit
```

**Comment by mhils** [comment]
Thanks for the report!

refs #1775

**Comment by mhils** [comment]
The sans-io implementation has been fixed in 81ed5c4, thanks again!
